src/Alert.php

```diff
@@ -0,0 +1,644 @@
+    {
+        if (empty($value)) {
+            throw new InvalidArgumentException('Header tag must be a string and cannot be empty.');
+        }
+
+        $new = clone $this;
+        $new->headerTag = $value;
+        return $new;
+    }
+
+    /**
+     * The attributes for rendering the `<i>` tag for icons alerts.
+     *
+     * @param array $value
+     *
+     * @return static
+     *
+     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function iconAttributes(array $value): self
+    {
+        $new = clone $this;
+        $new->iconAttributes = $value;
+        return $new;
+    }
+
+    /**
+     * Set icon css class in the alert component.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconClass(string $value): self
+    {
+        $new = clone $this;
+        $new->iconClass = $value;
+        return $new;
+    }
+
+    /**
+     * The attributes for rendering the container `<i>` tag.
+     *
+     * The rest of the options will be rendered as the HTML attributes of the `<i>` tag.
+     *
+     * @param array $value
+     *
+     * @return static
+     *
+     * {@see Html::renderTagAttributes()} for details on how attributes are being rendered.
+     */
+    public function iconContainerAttributes(array $value): self
+    {
+        $new = clone $this;
+        $new->iconContainerAttributes = $value;
+        return $new;
+    }
+
+    /**
+     * The CSS class for the container `<i>` tag.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconContainerClass(string $value): self
+    {
+        $new = clone $this;
+        $new->iconContainerClass = $value;
+        return $new;
+    }
+
+    /**
+     * Set icon text in the alert component.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function iconText(string $value): self
+    {
+        $new = clone $this;
+        $new->iconText = $value;
+        return $new;
+    }
+
+    /**
+     * Set layout the alert panel body.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function layoutBody(string $value): self
+    {
+        $new = clone $this;
+        $new->layoutBody = $value;
+        return $new;
+    }
+
+    /**
+     * Set layout the alert panel header.
+     *
+     * @param string $value
+     *
+     * @return static
+     */
+    public function layoutHeader(string $value): self
+    {
+        $new = clone $this;
+        $new->layoutHeader = $value;
+        return $new;
+    }
+
+    protected function run(): string
+    {
+        $new = clone $this;
+        return $new->renderAlert($new);
+    }
+
+    /**
+     * Render Alert.
+     */
+    private function renderAlert(self $new): string
+    {
+        $new->attributes['role'] = 'alert';
+        $new->attributes['id'] ??= $new->id ?? Html::generateId('alert-');
```

Comment:
Do we need to modify attributes here? These are used for rendering only so why setting them?

Reply:
There we are generating an id automatically if the user does not configure it, so we do in all the widgets.

Reply:
Generation itself if OK but we don't have to write it back to the widget attriubute.

Reply:
fixed.